include/ginkgo/core/log/logger.hpp

```diff
@@ -477,16 +465,71 @@ public:                                                              \
      * @param x  the solution vector (optional)
      * @param tau  the residual norm (optional)
      * @param implicit_tau_sq  the implicit residual norm squared (optional)
+     *
+     * @warning This on_iteration_complete function that this macro declares is
+     * deprecated. Please use the version with the stopping information.
      */
     [[deprecated(
-        "Please use the version with the additional status and stopped "
-        "parameter.")]] virtual void
+        "Please use the version with the additional stopping "
+        "information.")]] virtual void
     on_iteration_complete(const LinOp* solver, const size_type& it,
                           const LinOp* r, const LinOp* x, const LinOp* tau,
                           const LinOp* implicit_tau_sq) const
     {
-        this->on_iteration_complete(solver, nullptr, x, it, r, tau,
-                                    implicit_tau_sq, nullptr, false);
+#if defined(__GNUC__) || defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 5211, 4973, 4974)
+#endif
+        this->on_iteration_complete(solver, it, r, x, tau);
```

Comment:
calling a deprecated function from a deprecated function should not create a warning. Does it create one here?

Reply:
It creates a warning on gcc-13, so I will keep it.